app/assets/javascripts/sample_transfer.js.jsx

```diff
@@ -0,0 +1,135 @@
+var SampleTransfer = React.createClass({
+
+  getInitialState: function() {
+    return {
+      title: 'Transfer samples'
+    };
+  },
+
+  openTransferModal: function() {
+    this.refs.transferModal.show();
+    event.preventDefault();
+  },
+
+  closeTransferModal: function() {
+    this.refs.transferModal.hide();
+  },
+
+  render: function() {
+    return (<span>
+      <div id="modal_opener"  onClick={this.openTransferModal} >
+      </div>
+      <Modal ref="transferModal">
+        <h1>{this.state.title}</h1>
+        <SampleTransferModal onFinished={this.closeTransferModal} context={this.props.context} institutions={this.props.institutions} samples={this.props.samples} />
+      </Modal>
+    </span>);
+  }
+});
+
+var SampleTransferModal = React.createClass({
+  getInitialState: function() {
+    return {
+      institutionId: null,
+      includeQcInfo: null,
+      selectedSamples: selectedSamplesIds()
+    };
+  },
+
+  closeModal: function() {
+    event.preventDefault();
```

Comment:
This appears to be working fine in Chrome. But it errors in Firefox (97.0.1).
TBH I can't see what value `event` should be here. Where does that variable come from?

Reply:
Nice catch! Error is fixed in the last commit. The closeModal function is being used after the samples are done being transferred, but also when user clicks on the 'Cancel' button. Event is defined in the latter but not in the first case, since there is no user event being triggered.